akaze/src/image.rs

```diff
@@ -61,7 +61,17 @@ impl GrayFloatImage {
                     Luma([f32::from(gray_image[(x, y)][0]) / 65535f32])
                 })
             }
-            _ => unreachable!(),
+            other_image => {
```

Comment:
So looking at this whole block I'm not fully sure it would work as expected. The `DynamicImage` type has the following other variants:

* `ImageLumaA8`
* `ImageRgb8`
* `ImageRgba8`
* `ImageLumaA16`
* `ImageRgb16`
* `ImageRgba16`
* `ImageRgb32F`
* `ImageRgba32F`

The code below looks like it should work fine for Rgb8, but Rgb16 will need scaling by 65535, and Rgb32f won't need scaling as it should be in the range of 0.0-1.0 already. 

Also there's the question of handling images with alpha channels - but it may be that the `to_luma` function from `Image` has a handling of it that's sane I keep scouring the docs and I cant' find any reasonable explanation of what it will do and the trait implementations are done via macros and using num-traits so parsing the source is a bit painful.

If there's a way to get the scaling factor from the `DynamicImage` type this can be fixed without much of a change (ignoring the alpha issue under the assumption image can handle it well). If not there'll have to be a few more match arms for the different types. Though it may be worth special-casing the float ones to avoid an essentially no-op division

Reply:
This is now corrected!